packages/ui/src/machines/authenticator/defaultServices.ts

```diff
@@ -14,13 +14,26 @@ export const defaultServices = {
   // Validation hooks for overriding
   async validateCustomSignUp(formData): Promise<ValidatorResult> {},
   async validateConfirmPassword<Validator>(formData): Promise<ValidatorResult> {
-    const { password, confirm_password } = formData;
+    const {
+      password,
+      confirm_password,
+      touched_confirm_password,
+      touched_password,
+    } = formData;
 
     if (!password && !confirm_password) {
       // these inputs are clean, don't complain yet
       return null;
-    } else if (password && confirm_password && password !== confirm_password) {
-      // only return an error if both fields have text entered and the passwords do not match
+    } else if (
+      password &&
+      confirm_password &&
+      password !== confirm_password &&
+      ((touched_confirm_password && touched_password) ||
+        (password.length >= 6 && confirm_password.length >= 6))
+    ) {
+      // Only return an error if both fields have text entered,
+      // the passwords do not match, and the fields have been
+      // touched or the password and confirm password is longer then or equal to 6.
```

Comment:
Correct, so if they don't type anything in the second field, the error will not show. But if they tried to submit an error would show. 

I could add a check if they typed something in the password filed `abcd` and then clicked the `confirm_pasword` so at least it's touched, then clicked elsewhere it would show an error.

Reply:
OK Pushed up another check for that scenario